enigma-tools-u/src/web3_utils/raw_transaction.rs

```diff
@@ -0,0 +1,156 @@
+use ethereum_types::{H160, U256};
+use rlp::RlpStream;
+//use secp256k1::key::SecretKey;
+//use secp256k1::Message;
+//use secp256k1::Secp256k1;
+use tiny_keccak::keccak256;
+
+/// Description of a Transaction, pending or in the chain.
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+pub struct RawTransaction {
+    /// Nonce
+    pub nonce: U256,
+    /// Recipient (None when contract creation)
+    pub to: Option<H160>,
+    /// Transfered value
+    pub value: U256,
+    /// Gas Price
+    #[serde(rename = "gasPrice")]
+    pub gas_price: U256,
+    /// Gas amount
+    pub gas: U256,
+    /// Input data
+    pub data: Vec<u8>,
+}
+
+impl RawTransaction {
+    /// Signs and returns the RLP-encoded transaction
+//    pub fn sign(&self, private_key: &H256, chain_id: &i32) -> Vec<u8> {
+//        let hash = self.hash(*chain_id);
+//        let sig = ecdsa_sign(&hash, &private_key.0, &chain_id);
+//        self.raw_sign(sig.r, sig.s, sig.v)
+//    }
+
+    /// Signs and returns the RLP-encoded transaction - used to manually inject signature
+    pub fn raw_sign(&self, r: Vec<u8>, s: Vec<u8>, v: u64) -> Vec<u8> {
+        let mut r_n = r;
+        let mut s_n = s;
+        while r_n[0] == 0 {
+            r_n.remove(0);
+        }
+        while s_n[0] == 0 {
+            s_n.remove(0);
+        }
```

Comment:
It will be much more efficient to .find() the first non-0 byte, and slice the vector from there.

Reply:
Same as above